python/pydantic_core/core_schema.py

```diff
@@ -744,7 +744,7 @@ def decimal_schema(
 
 class StringSchema(TypedDict, total=False):
     type: Required[Literal['str']]
-    pattern: str
+    pattern: Any  # str | Pattern[str], but typing.Pattern[str] isn't recognized
```

Comment:
Define "isn't recognized" here - needs support in `generate_self_schema` maybe?

Reply:
Should be possible to add in this PR; though `generate_self_schema` is a pile of fun metaprogramming 😬